### PR TITLE
fix(search): close icon persists when field is empty (FE-2638)

### PR DESCRIPTION
**Pull request by grabkowski**
### Proposed behaviour
Close icon should be replaced by the search icon when the text is cleared.
image

### Current behaviour
When the text field is cleared using the close button, the close button is not replaced by the search icon.
image

### Checklist

- [x] Release notes (Conventional Commits) 
~~- [ ] Unit tests~~
- [x] Cypress automation tests
~~- [ ] Storybook added or updated~~
~~- [ ] Theme support~~
~~- [ ] Typescript `d.ts` file added or updated~~

### Testing instructions
1. run: npm start
2. go to: http://localhost:9001/?path=/story/test-search--basic
3. uncheck the `searchButton` knob
4. type `test` in the text field
5. click on the `close` button in the field

### CODE REVIEW

File: src/__experimental__/components/search/search.component.js [hunk 1]
```diff
@@ -17,9 +17,25 @@ const Search = ({
   );
   let inputRef = useRef(null);
   const [searchValue, setSearchValue] = useState(initialValue);
-  const [isActive, setIsActive] = useState(inputRef.current === document.activeElement);
+  const [isFocused, setIsFocused] = useState(inputRef.current === document.activeElement);
```
Comment (mkrds) [inline src/__experimental__/components/search/search.component.js]: I think there's no need for this `inputRef.current === document.activeElement` comparison as `inputRef.current` is undefined when `useState` is initialized anyway

File: src/__experimental__/components/search/search.component.js [hunk 2]
```diff
@@ -17,9 +17,25 @@ const Search = ({
   );
   let inputRef = useRef(null);
```
Comment (mkrds) [inline src/__experimental__/components/search/search.component.js]: Is seems that this ref is not used anywhere

Comment (grabkowski) [reply src/__experimental__/components/search/search.component.js]: removed

File: src/__experimental__/components/search/search.component.js (hunk 1 above)
Comment (grabkowski) [reply src/__experimental__/components/search/search.component.js]: you're right, removed

**Pull request merged**